lib/spack/spack/cmd/install.py

```diff
@@ -31,10 +32,64 @@
 level = "short"
 
 
+def parse_use_buildcache(args):
+    bc_parsed = {}
+    bc_keys = ["package", "dependencies", ""]
+    bc_values = ["only", "never", "auto"]
+    kv_list = re.findall("([a-z]+)?:?([a-z]+)", args.use_buildcache)
+
+    # Verify keys and values
+    bc_map = {k: v for k, v in kv_list if k in bc_keys and v in bc_values}
+    if not len(kv_list) == len(bc_map):
+        unrecognized_args = [(k, v) for k, v in kv_list if k not in bc_keys or v not in bc_values]
+        tty.error("Unrecognized arguments passed to use-buildcache: %s" % unrecognized_args)
+        tty.error(
+            "Expected: --use-buildcache "
+            ":{[auto|only|never],[[package:[auto|only|never]],[dependencies:[auto|only|never]]]"
+        )
+        exit(1)
+        return None
```

Comment:
Given the `exit(1)` just above this, I wouldn't think this `return` statement could ever be executed.

Reply:
ah, that isn't supposed to exit(1). I wasn't sure how to handle errors correctly with parsing.

Reply:
Normally, from spack commands, you can just return the value you want to be used as the exit code.  Probably worth verifying that value it makes it all the way back to the shell though.  Unless your test method already validates the exit code, in which case it will let you know.